app/helpers/application_helper/button/basic.rb

```diff
@@ -64,6 +65,9 @@ def skipped?
 
   # Tells whether the button should displayed in the toolbar or not
   def visible?
+    # hide select power management buttons for Openstack::InfraManager
+    return false if %w(host_standby host_shutdown host_reboot host_reset).include?(@child_id) &&
+      @record.kind_of?(ManageIQ::Providers::Openstack::InfraManager)
```

Comment:
yep +1 for right button class, I think there are quite some specific ones. @romanblanco probably knows more.
And definitely dont got the `kind_of?` route. In case there is no `supports?(:reboot)` for the record, there probably is a `validate_reboot`